src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -415,6 +418,7 @@ public abstract class GenerateProtoTask extends DefaultTask {
     return flavors
   }
 
+  @TypeChecked(TypeCheckingMode.SKIP) // Don't depend on AGP
```

Comment:
I'd prefer to add facade on SourceSet in some reasons:
1 better code, no Object
2 agp contains two different classes for sourceSets, deprecated one and the new one. Its the main reason why plugin doesnt work with newer agp.
3 a bit easy support to kmp

Reply:
I have little issue with that, although it seems best to do that separately. Right now it is "get as much compiling static as easily possible." This is a trivial change and gets us closer to Java 11, if not AGP. Over time we  increase what is compiled static and clean AGP handling (and fix AGP).